File: project-euler.java/src/main/java/euler/solutions/_9.java
package euler.solutions;

import euler.common.BaseSolution;

public class _9 extends BaseSolution {

    @Override
    public long solve() {
        for (int a = 1; a <= 998; a++) {
            for (int b = a + 1; b <= 999 - a; b++) {
                int c = 1000 - a - b;
                if (a * a + b * b == c * c) {
                    return a * b * c;
                }
            }
        }
        throw new RuntimeException("Failed to solve problem");
    }
}

File: project-euler.go/solutions/_9/9.go
package _9

func Solve() int64 {
	for a := 1; a <= 998 ; a++ {
		for b := a + 1; b <= 999 - a; b++ {
			c := 1000 - a - b
			if (a * a + b * b == c * c) {
				return int64(a * b *c)
			}

		}
	}

	panic("Failed to solve problem")
}
